### ISSUE TITLE
Handle unmatched spans

### ISSUE THREAD

**Issue by allanj** [redthing1/layoutlm_experiments#1]
I ran your preprocessing script to get the dataset cache.

It seems for those answers that we cannot find matched spans, we just put a `start_position = 0` and `end_position=0`.

Any thought on doing this? rather than remove them

**Comment by allanj** [comment]
By the way, I got something similar here at the 4000th step checkpoint
```bash
num items: 5039
num exact match: 3510
exact acc: 69.66%
anls: 0.741
```
This number seems similar to what you reported here https://github.com/microsoft/unilm/issues/799

But it seems the 74.1 ANLS is calculated based on the number 5039.
```python3
print(f"num items: {num_items}")
print(f"num exact match: {num_exact_correct}")
print(f"exact acc: {num_exact_correct / num_items:.2%}")
anls = total_normalized_levenshtein / num_items
print(f"anls: {anls:.3f}")
```

But I think it should be divided by the number of all validation items, where the number should be 5349

**Comment by allanj** [comment]
I guess it should still generate/predict the answer span. but directly compare with the gold answers, despite we cannot find any matched spans (during validation.)

**Comment by redthing1** [comment]
I have been moving away from the XQA (extractive qa) formulation because it has several major issues that are not easy to resolve:

extractive qa predicts start and end positions of the answer within the input tokens, but this has many problems
- subword tokenization means you can't always express expected answers in model vocabulary, because the tokens are already given 
- if the answer is a price "$10" but this $10 exists in multiple places in the input text, when encoding the dataset pairs (which are provided as seq2seq) you lose information
- when selecting a token range, this means that all answer tokens must be consecutive, but this isn't always the case as it's unclear the order in which nearby tokens are tokenized; for example, you can have a street address that is one line after another, but if there's something else on the right side of the page then the street address parts are not consecutive and extraction of this is poorly defined for the model
- sometimes you want to ask questions that are not always extractive, like "is vegetarian" or "yes/no" 
- instead, doing seq2seq QA would be a much better approach as it allows free form abstractive answers

**Comment by allanj** [comment]
I also did that previously. But it seems the performance of seq2seq QA is quite worse than the extractive way on  DocVQA task

**Comment by redthing1** [comment]
What are you ANLS numbers for seq2seq? I have 74.9

**Comment by allanj** [comment]
I think that probably depends on the decoder that we use. I used a pretrained Bart Decoder and got around 72.5 ANLS. 
Batch size: 48, learning rate: 2e-5. 100 epochs

**Comment by allanj** [comment]
I wonder if you are able to get the OCR for the test set as well
